src/main/scala/org/constellation/domain/transaction/TransactionValidator.scala

```diff
@@ -4,21 +4,30 @@ import cats.data.ValidatedNel
 import cats.effect.Sync
 import cats.syntax.all._
 import org.constellation.schema.transaction.{LastTransactionRef, Transaction}
+import scodec.bits.Bases.Alphabets.Base58
 
 object TransactionValidator {
   type ValidationResult[A] = ValidatedNel[TransactionValidationError, A]
 
+  val base58Chars = (('1' to '9') ++ ('A' to 'Z') ++ ('a' to 'z'))
+    .filterNot(c => List('O', 'I', 'l').contains(c))
+    .toSet
+
   def validateSourceSignature(tx: Transaction): ValidationResult[Transaction] =
     if (tx.isValid) tx.validNel else InvalidSourceSignature(tx).invalidNel
 
   def validateEmptyDestinationAddress(tx: Transaction): ValidationResult[Transaction] =
     if (tx.dst.address.nonEmpty) tx.validNel else EmptyDestinationAddress(tx).invalidNel
 
-  def validateDestinationAddress(tx: Transaction): ValidationResult[Transaction] =
-    if (tx.dst.address.length > 30 && tx.dst.address.startsWith("DAG") && tx.src != tx.dst)
+  def validateDestinationAddress(tx: Transaction): ValidationResult[Transaction] = {
+    val par = tx.dst.address.substring(4).filter(Character.isDigit).map(_.toString.toInt).sum % 9
+    val base58 = tx.dst.address.forall(base58Chars.contains)
+
+    if (base58 && tx.dst.address.startsWith(s"DAG${par}") && tx.dst.address.length > 30 && tx.dst.address.startsWith("DAG") && tx.src != tx.dst)
```

Comment:
I would also restrict that it's exactly 40 chars, where first 3 are "DAG", then 1 par, and 36 base58